### PR TITLE
tests/kola: add test for systemd-timesyncd

### PR DESCRIPTION
**Pull request by dustymabe**

### CODE REVIEW

File: tests/kola/ntp/chrony/dhcp-propagation [hunk 1]
```diff
@@ -78,13 +79,12 @@ main() {
     # Name: time-c-g.nist.gov  IP address: 129.6.15.30
     NTPSERVER=""
```
Comment (jlebon) [inline tests/kola/ntp/chrony/dhcp-propagation]: Minor: I think we can drop this too now?

File: tests/kola/ntp/chrony/dhcp-propagation [hunk 2]
```diff
@@ -78,13 +79,12 @@ main() {
...
-    while [[ $retries -gt 0 && ! $NTPSERVER =~ "time-c-g.nist.gov" ]]; do
-         NTPSERVER=$(chronyc sources)
-         if [[ $NTPSERVER != *"time-c-g.nist.gov"* ]]; then
-             echo "waiting for ntp server to appear"
-             sleep 1
-             retries=$((retries - 1))
-         fi
+    while [[ $retries -gt 0 ]]; do
+         NTPSERVER=$(chronyc -n sources)
+         [[ "$NTPSERVER" =~ "$NTPHOSTIP" ]] && break
```
Comment (jlebon) [inline tests/kola/ntp/chrony/dhcp-propagation]: Feels like the commit message here is missing the rationale for why we've moving away from being name resolution-based. Did you observe a failure?

File: tests/kola/binfmt/qemu [hunk 3]
```diff
@@ -22,7 +22,7 @@ set -xeuo pipefail
 
 case "$(arch)" in
     aarch64|s390x)
-        containerArch=$(podman run --arch=amd64 --rm registry.fedoraproject.org/fedora:36 arch)
+        containerArch=$(podman run --arch=amd64 --rm registry.fedoraproject.org/fedora:37 arch)
```
Comment (jlebon) [inline tests/kola/binfmt/qemu:25]: Should we add a step for this in the rebase checklist?

File: tests/kola/ntp/chrony/dhcp-propagation [hunk 4]
```diff
@@ -1,96 +1,40 @@
...
-#     that.
-# - timeoutMin: 15
-#   - Pulling and building the container can take a long time if a
-#     slow mirror gets chosen.
-# - minMemory: 1536
-#   - There's a bug in dnf that is causing OOM on low memory systems:
-#     https://bugzilla.redhat.com/show_bug.cgi?id=1907030
-#     https://pagure.io/releng/issue/10935#comment-808601
+## kola: 
+##   # Add the needs-internet tag. This test builds a container from remot
```
Comment (jlebon) [inline tests/kola/ntp/chrony/dhcp-propagation]: remote

File: tests/kola/ntp/timesyncd/dhcp-propagation/config.bu [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    - name: systemd-timesyncd.service
+      enabled: true
+# There currently isn't any NetworkManager dispatcher that
+# propagates DHCP info to timesyncd in Fedora so we'll just
+# pull one that we can use from the internet.
+storage:
+  files:
+    - path: /etc/NetworkManager/dispatcher.d/30-timesyncd
+      contents:
+        source: "https://github.com/eworm-de/networkmanager-dispatcher-timesyncd/raw/master/30-timesyncd"
```
Comment (jlebon) [inline tests/kola/ntp/timesyncd/dhcp-propagation/config.bu:16]: I'd prefer just having it be inlined here to keep it self-contained. It's not very large. (With linkback of course.)

File: tests/kola/ntp/chrony/dhcp-propagation (hunk 1 above)
Comment (dustymabe) [reply tests/kola/ntp/chrony/dhcp-propagation]: yep

File: tests/kola/ntp/chrony/dhcp-propagation (hunk 2 above)
Comment (dustymabe) [reply tests/kola/ntp/chrony/dhcp-propagation]: basically it makes it easier for the code to support timesyncd in the following commits. I added a note to the commit message.

File: tests/kola/binfmt/qemu (hunk 3 above)
Comment (dustymabe) [reply tests/kola/binfmt/qemu:25]: probably

File: tests/kola/ntp/timesyncd/dhcp-propagation/config.bu (hunk 5 above)
Comment (dustymabe) [reply tests/kola/ntp/timesyncd/dhcp-propagation/config.bu:16]: ehh. I'm hesitant to copy other people's code around. What's the concern here? That the repo disappears?

If that happens then we should probably stop testing this.

Comment (jlebon) [reply tests/kola/ntp/timesyncd/dhcp-propagation/config.bu:16]: That the file moves, disappears, or is modified and that we're unnecessarily adding a dependency in our CI we could easily avoid.

Comment (dustymabe) [reply tests/kola/ntp/timesyncd/dhcp-propagation/config.bu:16]: Unfortunately the code at https://github.com/eworm-de/networkmanager-dispatcher-timesyncd doesn't have a license. I guess we could just not add the test - or rework it in some way.

Comment (dustymabe) [reply tests/kola/ntp/timesyncd/dhcp-propagation/config.bu:16]: I'm thinking if it disappears or starts breaking we can just delete it.

**Review by jlebon — APPROVED** [review]
Ack, sounds good!

**Comment by travier** [comment]
This is failing on RHCOS: https://github.com/openshift/os/pull/1088
Maybe this should be FCOS only?

**Comment by travier** [comment]
More precisely, `ext.config.shared.ntp.timesyncd.dhcp-propagation` is failing.

**Comment by dustymabe** [comment]
> This is failing on RHCOS: [openshift/os#1088](https://github.com/openshift/os/pull/1088) Maybe this should be FCOS only?

`ntp.timesyncd.dhcp-propagation` absolutely should be `fcos` only. An earlier version of this PR had it that way but then I lost it in a copy/paste.

PR to fix: 
https://github.com/coreos/fedora-coreos-config/pull/2127

**Pull request merged**